extensions/amp-access/0.1/amp-access.js

```diff
@@ -145,30 +152,52 @@ export class AccessService {
   }
 
   /**
+   * @return {!Promise<string>}
+   * @private
+   */
+  getReaderId_() {
+    if (!this.readerIdPromise_) {
+      // TODO(dvoytenko): Confirm no consent is needed. The assumtpion is that
+      // this is the case given that this is an essential part of access system.
+      const consent = Promise.resolve();
+      this.readerIdPromise_ = this.cid_.get('amp-access', consent);
+    }
+    return this.readerIdPromise_;
+  }
+
+  /**
+   * @param {string} url
+   * @return {!Promise<string>}
+   * @private
+   */
+  buildUrl_(url) {
+    return this.getReaderId_().then(readerId => {
+      return this.urlReplacements_.expand(url, {
+        'READER_ID': readerId
```

Comment:
Cool. This is what opt_data does in UrlReplacements. Could you add @param docs over there?